### PR TITLE
{Core} `doc_string_source` must be full path

### LINKED ISSUE #27793: `ModuleNotFoundError` warnings in `az self-test --debug`
### Describe the bug

Lots of `ModuleNotFoundError: No module named 'azure.mgmt.keyvault.v2023_02_01.azure'` errors are raised when running `az self-test`. This occurs as early as 2.30.0
```
cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
  File "c:\users\kk\developer\azure-cli\src\azure-cli-core\azure\cli\core\profiles\_shared.py", line 651, in _get_attr
    op = import_module(full_mod_path)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\kk\AppData\Local\Programs\Python\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1128, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1128, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1128, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1142, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'azure.mgmt.keyvault.v2023_02_01.azure'
```

In https://github.com/Azure/azure-cli/blob/68990c9fa3da380a2a30dde14527999876844fe8/src/azure-cli-core/azure/cli/core/__init__.py#L709,
if `doc_string_source` is `azure.mgmt.keyvault.v2023_02_01.models#VaultProperties`, when it calls https://github.com/Azure/azure-cli/blob/68990c9fa3da380a2a30dde14527999876844fe8/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L646-L651,
the `full_mod_path` become `azure.mgmt.keyvault.v2023_02_01.azure.mgmt.keyvault.v2023_02_01.models`, and it fails to import.

### PR DESCRIPTION
**Pull request by bebound**
**Description**

This PR fixes the errors when running `az self-test --debug`
The `doc_string_source` in `commands.py` must be full path now.

Before this PR, in https://github.com/Azure/azure-cli/blob/1fdda16d0812ab52cb5079802588fecafa27d669/src/azure-cli-core/azure/cli/core/__init__.py#L707-L718
1. If `doc_string_source` is `azure.mgmt.netapp.models#Snapshot`, `APIVersionException` is raised. Run line 718
2. if `doc_string_source` is `azure.mgmt.keyvault.v2023_02_01.models#VaultProperties`, it tries to import `azure.mgmt.keyvault.v2023_02_01.azure.mgmt.keyvault.v2023_02_01.models` in `_get_attr()` and fails. Run line 718
3. If `doc_string_source` is `file#FileService.list_directories_and_files` in azure stack, it finally imports `azure.multiapi.storage.v2017_04_17.file#FileService.list_directories_and_files`

In case2, lots of `ModuleNotFoundError: No module named 'azure.mgmt.keyvault.v2023_02_01.azure'` is raised.

In CLI and extension, the `doc_string_source` is full path except azure stack. I change the `doc_string_source` to full path, so CLI does not need to call `get_models` to concat the `sdk_path` and `class_path` and `model = self.get_models(doc_string_source)` can be eliminated.

Close #27793

**Testing Guide**
Run `az self-test --debug` to check the output

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: src/azure-cli-core/azure/cli/core/__init__.py [hunk 1]
```diff
@@ -704,21 +704,15 @@ def _apply_doc_string(self, dest, command_kwargs):
...
+        (path, model_name) = doc_string_source.split('#', 1)
+        method_name = None
+        if '.' in model_name:
+            (model_name, method_name) = model_name.split('.', 1)
+        module = import_module(path)
+        model = getattr(module, model_name)
+        if method_name:
+            model = getattr(model, method_name, None)
         if not model:
             raise CLIError("command authoring error: source '{}' not found.".format(doc_string_source))
```
Comment (bebound) [inline src/azure-cli-core/azure/cli/core/__init__.py:717]: This line is not reachable?

File: src/azure-cli-core/azure/cli/core/__init__.py [hunk 2]
```diff
@@ -704,23 +704,15 @@ def _apply_doc_string(self, dest, command_kwargs):
...
-            (path, model_name) = doc_string_source.split('#', 1)
-            method_name = None
-            if '.' in model_name:
-                (model_name, method_name) = model_name.split('.', 1)
-            module = import_module(path)
-            model = getattr(module, model_name)
-            if method_name:
-                model = getattr(model, method_name, None)
-        if not model:
-            raise CLIError("command authoring error: source '{}' not found.".format(doc_string_source))
```
Comment (bebound) [inline src/azure-cli-core/azure/cli/core/__init__.py:723]: `model` can't be empty, this code is unreachable.

**Comment by bebound** [comment]
Solved by https://github.com/Azure/azure-cli/pull/27848

**Pull request closed without merging**